### PR TITLE
Add MovementAction to Cpt. Toad

### PR DESCRIPTION
**Pull request by andreafra**
Cpt. Toad moves relatively to the camera position and angle

Also removed a couple of debug leftovers

### CODE REVIEW

**Comment by andreafra** [comment]
It'd be nice to make the Toad rotate using the `alpha` angle in his Movement action.

File: src/engine/Core.ts [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { Light } from "./Lights";
 import { Node, RenderNode, State } from "./SceneGraph";
 import * as DebugLine from "./debug/Lines";
 import { WebGLProgramInfo } from "./Shaders";
+import { GetMode } from "../main";
```
Comment (aleasto) [inline src/engine/Core.ts]: No

File: src/engine/Core.ts [hunk 2]
```diff
@@ -51,7 +52,7 @@ function Render(time: DOMHighResTimeStamp) {
 		renderFunction();
 	}
 
-	DebugLine.Render(viewProjectionMatrix);
+	/* if (GetMode() === "EDITOR") */ DebugLine.Render(viewProjectionMatrix);
```
Comment (aleasto) [inline src/engine/Core.ts]: No

File: src/engine/SceneGraph.ts [hunk 3]
```diff
@@ -4,6 +4,7 @@ import * as Engine from "./Core";
 import { Light } from "./Lights";
 import { WebGLProgramInfo } from "./Shaders";
 import { gl } from "./Core";
+import { GetMode } from "../main";
```
Comment (aleasto) [inline src/engine/SceneGraph.ts]: No

File: src/engine/SceneGraph.ts [hunk 4]
```diff
@@ -89,13 +90,14 @@ export class Node<T extends State> {
...
-	AddAction(action: (state: T) => void) {
+	AddAction(action: (deltaTime: number, state: T) => void) {
 		this.actions.push(action);
 	}
 
 	Update(deltaTime: number, VPMatrix: number[], worldMatrix?: number[]) {
 		this.UpdateWorldMatrix(worldMatrix);
-		this.ExecuteActions();
+
+		if (GetMode() === "GAME") this.ExecuteActions(deltaTime);
```
Comment (aleasto) [inline src/engine/SceneGraph.ts]: No

File: src/models/Toad.ts [hunk 5]
```diff
@@ -104,9 +111,41 @@ export function Spawn() {
...
+	toadNode.AddAction(MovementAction);
+
 	// Set relationships between nodes
 	headLight.SetParent(toadNode);
 	toadNode.SetParent(Engine.ROOT_NODE);
 
 	return toadNode;
 }
+
+const MovementAction = (deltaTime: number, state: ToadState): void => {
```
Comment (aleasto) [inline src/models/Toad.ts:125]: if (GetMode() != "GAME") return;

**Review by aleasto — CHANGES_REQUESTED** [review]

File: src/models/Toad.ts [hunk 6]
```diff
@@ -104,9 +111,41 @@ export function Spawn() {
...
+	let position = utils.ComputePosition(state.localMatrix, [0, 0, 0]);
```
Comment (aleasto) [inline src/models/Toad.ts]: worldMatrix?

**Review by aleasto — APPROVED** [review]

**Pull request merged**